usr/sbin/pkcstok_migrate/pkcstok_migrate.c

```diff
@@ -127,7 +164,7 @@ static CK_RV migrate_public_token_object(const char *data_store, const char *nam
                                          unsigned char *data, unsigned long len)
 {
     const char *tokobj = "TOK_OBJ";
-    char fname[PATH_MAX + 1 + strlen(tokobj) + 1 + strlen(name) + 1];
+    char fname[PATH_MAX];
```

Comment:
I agree. In the original code this ugly length calculation was necessary to avoid compiler warnings.

Reply:
Well, if a path that long is needed, fopen will return ENAMETOOLONG.  So no point in using more memory and a dynamically-sized array.